packages/api/src/providers/fitbit.ts

```diff
@@ -373,7 +388,86 @@ export class Fitbit extends Provider implements OAuth2 {
     );
   }
 
-  // Creates a subscription for all or one specific collectionType
+  // Finds existing users connected to this Fitbit account, revokes their tokens and sends WH updates about the disconnections
+  async revokeTokenFromExistingUsers(
+    userId: string,
+    fitbitUserId: string,
+    activeSubscriptions: string[],
+    accessToken: string
+  ) {
+    try {
+      if (activeSubscriptions.length > 0) {
+        const connectedUsers = await getConnectedUsersByTokenOrFail(
+          ProviderSource.fitbit,
+          fitbitUserId
+        );
+        await Promise.all(
+          Object.values(connectedUsers).map(async user => {
+            if (user.dataValues.id !== userId) {
+              await this.oauth.revokeLocal(user);
+              user = await getConnectedUserOrFail({
+                id: user.id,
+                cxId: user.cxId,
+              });
+              sendProviderDisconnected(user, [ProviderSource.fitbit]);
+            }
+          })
+        );
+        await this.deleteSubscriptions(activeSubscriptions, fitbitUserId, accessToken);
+      }
+    } catch (err) {
+      console.log("Failed to revoke the token of a Fitbit user.", err);
```

Comment:
let's not pass error as an independent parameter to `console.[log|error]()`... this results in NodeJS printing the stack trace of the error in multiple lines, making logs on AWS CloudWatch super noisy and hard to read (there's a ticket to fix it, https://github.com/metriport/metriport-internal/issues/693).

Better to either wrap it with `String(err)` (safe to pass as a separated param) or pass `err` as a param to a template string, like

```ts
`Failed..., error: ${err}`
```

See `sendProviderDisconnected()` for an example.